server/src/lib/model/db/mysql.ts

```diff
@@ -22,55 +21,45 @@ export type MysqlOptions = {
 };
 
 // Default configuration values, notice we dont have password.
-const DEFAULTS = {
+const DEFAULTS: MysqlOptions = {
   user: 'voiceweb',
   database: 'voiceweb',
   password: '',
   host: 'localhost',
   port: 3306,
   max: 10,
   idleTimeoutMillis: 30000,
+  multipleStatements: false,
 };
 
 export default class Mysql {
+  config: CommonVoiceConfig;
   options: MysqlOptions;
   conn: IConnection;
   rootConn: IConnection;
 
   /**
    * Get options from params first, then config, and falling back to defaults.
    *   For configuring, use the following order of priority:
-   *     1. passed in options
-   *     2. options in config.json
-   *     3. hard coded DEFAULTS
+   *     1. options in config.json
+   *     2. hard coded DEFAULTS
    */
-  private getFullOptions(options?: MysqlOptions): MysqlOptions {
-    options = options || Object.create(null);
+  private getMysqlOptions(config: CommonVoiceConfig): MysqlOptions {
     return {
-      user: getFirstDefined(options.user, config.MYSQLUSER, DEFAULTS.user),
-      database: getFirstDefined(
-        options.database,
-        config.MYSQLDBNAME,
-        DEFAULTS.database
-      ),
-      password: getFirstDefined(
-        options.password,
-        config.MYSQLPASS,
-        DEFAULTS.password
-      ),
-      host: getFirstDefined(options.host, config.MYSQLHOST, DEFAULTS.host),
-      port: getFirstDefined(options.port, config.MYSQLPORT, DEFAULTS.port),
-      max: getFirstDefined(options.max, DEFAULTS.max),
-      idleTimeoutMillis: getFirstDefined(
-        options.idleTimeoutMillis,
-        DEFAULTS.idleTimeoutMillis
-      ),
+      user: getFirstDefined(config.MYSQLUSER, DEFAULTS.user),
```

Comment:
While I know JS truthiness is not to be trusted, for cases where I know that something falsy wouldn't be passed (like here), I'd personally go with: `config.MYSQLUSER || DEFAULTS.user`

Reply:
I agree with you in principal, but to me consistency is nice here since it's easier to maintain down the road. This particular piece of code has already been touched too many times due to JS truthiness problems, and each time I have to touch it I have to *think*, which nobody likes doing.

Reply:
if only we could write some program to automatically _test_ if it's working correctly!

Reply:
@nevik *poof* now we do!

https://github.com/mozilla/voice-web/blob/2797248a1a4eac737cbdd9fbe8242b5049bc248a/server/src/test/schema-migration.ts#L11

#magic :)